packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -1,60 +1,107 @@
   OpenAiEmbeddingOutput,
-  OpenAiChatCompletionOutput
+  OpenAiChatCompletionOutput,
+  OpenAiChatModel,
+  OpenAiEmbeddingModel
 } from './openai-types.js';
 
 const apiVersion = '2024-02-01';
 
 /**
- * OpenAI GPT Client.
+ * OpenAI Client.
  */
-export class OpenAiClient implements BaseClient<BaseLlmParameters> {
+export class OpenAiClient {
   /**
    * Creates a completion for the chat messages.
+   * @param model - The model to use for the chat completion.
    * @param data - The input parameters for the chat completion.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
+    model: OpenAiChatModel | { name: OpenAiChatModel; version: string },
     data: OpenAiChatCompletionParameters,
+    deploymentResolver?: DeploymentResolver,
     requestConfig?: CustomRequestConfig
   ): Promise<OpenAiChatCompletionOutput> {
+    const deploymentId = await resolveOpenAiDeployment(
+      model,
+      deploymentResolver
+    );
     const response = await executeRequest(
       {
-        url: `/inference/deployments/${data.deploymentConfiguration.deploymentId}/chat/completions`,
+        url: `/inference/deployments/${deploymentId}/chat/completions`,
         apiVersion
       },
       data,
-      requestConfig
+      mergeRequestConfig(requestConfig)
     );
     return response.data;
   }
   /**
    * Creates an embedding vector representing the given text.
-   * @param data - The input parameters for the chat completion.
+   * @param model - The model to use for the embedding computation.
+   * @param data - The text to embed.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async embeddings(
+    model:
+      | OpenAiEmbeddingModel
+      | { name: OpenAiEmbeddingModel; version: string },
     data: OpenAiEmbeddingParameters,
+    deploymentResolver?: DeploymentResolver,
     requestConfig?: CustomRequestConfig
   ): Promise<OpenAiEmbeddingOutput> {
+    const deploymentId = await resolveOpenAiDeployment(
+      model,
+      deploymentResolver
+    );
     const response = await executeRequest(
-      {
-        url: `/inference/deployments/${data.deploymentConfiguration.deploymentId}/embeddings`,
-        apiVersion
-      },
+      { url: `/inference/deployments/${deploymentId}/embeddings`, apiVersion },
       data,
-      requestConfig
+      mergeRequestConfig(requestConfig)
     );
     return response.data;
   }
 }
+
+async function resolveOpenAiDeployment(
+  model: string | { name: string; version: string },
+  resolver?: DeploymentResolver
+) {
+  if (typeof resolver === 'string') {
+    return resolver;
+  }
+  const llm =
+    typeof model === 'string' ? { name: model, version: 'latest' } : model;
+  const deployment = await resolveDeployment({
```

Comment:
if `resolver` is a function it is currently ignored. See https://github.tools.sap/AI/gen-ai-hub-sdk-js-backlog/issues/92